core/src/main/scala/pureconfig/error/ConfigReaderFailure.scala

```diff
@@ -298,6 +299,16 @@ final case class NoValidCoproductChoiceFound(value: ConfigValue, location: Optio
 }
 
 /**
+ * A failure representing the inability to read a requested file.
+ */
+final case class CannotReadFile(path: Path) extends ConfigReaderFailure {
```

Comment:
Yeah, I thought about that too. And there are more reasons why you might not be able to read a file, such as the case where a file is locked by the OS.

In this PR I didn't focus in giving the user information about exactly why the file cannot be read, but I think that would be better passed along to the user by making a general failure like `CannotReadFile` hold a `reason` string than by creating a different `Failure` for each type of IO error.